src/matrix_multiply.jl

```diff
@@ -284,8 +289,28 @@ end
     end
 end
 
-
-@generated function mul_blas!(::Size{s}, c::StaticMatrix{<:Any, <:Any, T}, ::Size{sa}, ::Size{sb}, a::StaticMatrix{<:Any, <:Any, T}, b::StaticMatrix{<:Any, <:Any, T}) where {s,sa,sb, T <: BlasFloat}
+@inline mul_blas!(Sc::Size{s}, c::SizedArray{<:Any,T},
+        Sa::Size{sa}, Sb::Size{sb},
+        a::SizedArray{<:Any,T}, b::SizedArray{<:Any,T},
+        α::Real=one(T), β::Real=zero(T)) where {s,sa,sb, T <: BlasFloat} =
+    BLAS.gemm!('N','N', α, a.data, b.data, β , c.data)
+
+@inline mul_blas!(Sc::Size{s}, c::SizedArray{<:Any,T},
+        Sa::Size{sa}, Sb::Size{sb},
+        a::Transpose{<:Any, <:SizedArray}, b::SizedArray{<:Any,T},
+        α::Real=one(T), β::Real=zero(T)) where {s,sa,sb, T <: BlasFloat} =
+    BLAS.gemm!('T','N', α, a.parent.data, b.data, β , c.data)
+
+@inline mul_blas!(Sc::Size{s}, c::StaticMatrix{<:Any, <:Any, T},
+        Sa::Size{sa}, Sb::Size{sb},
+        a::Transpose{<:Any, <:StaticMatrix}, b::StaticMatrix{<:Any, <:Any, T},
+        α::Real=one(T), β::Real=zero(T)) where {s,sa,sb, T <: BlasFloat} =
+    BLAS.gemm!('T','N', α, a.parent, b, β , c)
```

Comment:
This rather a lot of duplication which will be hard to maintain.

I think all you need is a utility function to "get the data pointer and transposed-ness" out of `a`, `b`, `c`? Then you can collapse all these down again into one function.